### ISSUE TITLE
MvxIoCResolveException Exception when back button clicked

### ISSUE THREAD

**Issue by NPadrutt** [MvvmCross/MvvmCross#2984]
Since I upgraded my Xamarin Forms app to MvvmCross v6.1.2 I get always this exception on Android when I click on the back button of the device. The Button of the navigation bar works fine. Forward navigation is fine as well.

```
MvvmCross.Exceptions.MvxIoCResolveException: Failed to resolve type MvvmCross.Forms.Presenters.IMvxFormsPagePresenter
```
In this branch the error occurs: https://github.com/NPadrutt/MoneyFox.Windows/tree/mvvmcrossBackButtonIssue

**Comment by Goncharuk-Nikita** [comment]
Yea, i also get these exception.

**Comment by MartinZikmund** [comment]
This fails because `IMvxFormsPagePresenter` is not registered in the default setup. In the meantime you can use a workaround - register the `IMvxFormsPagePresenter` that is created manually in your Android `Setup` class:

```
protected override IMvxFormsPagePresenter CreateFormsPagePresenter(IMvxFormsViewPresenter viewPresenter)
{
    var formsPresenter = base.CreateFormsPagePresenter(viewPresenter);
    Mvx.RegisterSingleton(formsPresenter);
    return formsPresenter;
}

```

**Comment by NPadrutt** [comment]
Thanks @MartinZikmund. The workaround works.

**Comment by nmilcoff** [comment]
Hey @nickrandolph, am I right if I say this was solved in #2972?

**Comment by nickrandolph** [comment]
Yes I do believe so. Let's wait to see the next release and then close once verified
